### PR TITLE
Set same site to None when `new_header` is off

### LINKED ISSUE #2029: UI cookies are rejected when in the iframe
## Description

Layout cookies are rejected when inside the iframe.

## Reproduction

1. Go to https://wp.org/openverse
2. Open the developer tools console.
3. See error: several warnings about the cookies being rejected like this one:
``` “uiInstructionsSnackbarState” has been rejected because it is in a cross-site context and its “SameSite” is “Lax” or “Strict”.```

## Additional context

We should set the cookies SameSite to `None` when the `new_header` flag is off. Since we do not store any sensitive information in the cookies, it will not cause any security problems. However, it's still better to use a stricter "SameSite" policy when we are out of the iframe.

It is probably the easiest to add the fix here:
https://github.com/WordPress/openverse-frontend/blob/84b9f8162fe653d6d869c38470b9f12a7589fca7/src/stores/ui.ts#LL161C33-L161C33

We need to add a check for `new_header` and update the `opts` (setting `SameSite` to `None` if the `new_header` is `off`).

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #2029 by @obulat 

## Description

This PR sets the UI cookies' `sameSite` to `None` if the new_header is Off to stop them from being rejected when Openverse is inside the iframe.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/stores/ui.ts [hunk 1]
```diff
@@ -159,6 +160,10 @@ export const useUiStore = defineStore('ui', {
 
     updateCookies() {
       const opts = cookieOptions
+      if (useFeatureFlagStore().isOn('new_header')) {
+        opts.sameSite = 'none'
+      }
+
```
Comment (zackkrida) [inline src/stores/ui.ts:166]: I think this may need some cleanup. Is `opts` being used anywhere?

**Review by zackkrida — CHANGES_REQUESTED** [review]
I'm seeing some issues locally where only the `uiBreakpoint` cookie is getting `none` set correctly.

CleanShot 2022-12-06 at 09 31 54

Also mentioned some potential unused code inline in the comments.

File: src/stores/ui.ts (hunk 1 above)
Comment (obulat) [reply src/stores/ui.ts:166]: Yes, it's used in the `$cookies.setAll` call below. But it wasn't setting the options correctly.

**Review by zackkrida — APPROVED** [review]
The logic looks correct now; LGTM!

**Review by dhruvkb — APPROVED** [review]
LGTM!

**Pull request merged**